tsrc/git.py

```diff
@@ -113,13 +113,94 @@ def reset(repo, ref):
     run_git(repo, "reset", "--hard", ref)
 
 
-def get_status(working_path):
+def get_complete_status(working_path):
     _, out = run_git(working_path, "status", "--porcelain", raises=False)
+
+    untracked = 0
+    staged = 0
+    not_staged = 0
+    added = 0
+
     for line in out.splitlines():
         if line.startswith("??"):
-            return "untracked files"
+            untracked = untracked + 1
         if line.startswith(" M"):
-            return "modified files"
+            staged = staged + 1
+        if line.startswith(" .M"):
+            not_staged = not_staged + 1
         if line.startswith("A "):
-            return "non-committed files"
-    return "clean"
+            added = added + 1
+
+    return untracked, staged, not_staged, added
+
+
+def git_remote_status(working_path):
+    _, ahead_rev = run_git(working_path, "rev-list", "@{upstream}..HEAD", raises=False)
+    ahead = len(ahead_rev.splitlines())
+
+    _, behind_rev = run_git(working_path, "rev-list", "HEAD..@{upstream}", raises=False)
+    behind = len(behind_rev.splitlines())
+
+    ahead_symbol = "↑"
+    behind_symbol = "↓"
+    diverge_symbol = "↕"
+
+    if (ahead == 0) and (behind == 0):
```

Comment:
There's a lot of code to deal with the pluralization of the 'commit' word here. I believe it could be extracted into its own function.